x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -0,0 +1,290 @@
+          body: { message: err.message },
+        });
+      }
+      return response.internalError({ body: err });
+    }
+  };
+};
+
+export async function getHostData(
+  metadataRequestContext: MetadataRequestContext,
+  id: string,
+  queryStrategyVersion?: MetadataQueryStrategyVersions
+): Promise<HostInfo | undefined> {
+  const queryStrategy = await metadataRequestContext.endpointAppContextService
+    ?.getMetadataService()
+    ?.queryStrategy(
+      metadataRequestContext.requestHandlerContext.core.savedObjects.client,
+      queryStrategyVersion
+    );
+
+  const query = getESQueryHostMetadataByID(id, queryStrategy!);
+  const hostResult = queryStrategy!.queryResponseToHostResult(
+    await metadataRequestContext.requestHandlerContext.core.elasticsearch.legacy.client.callAsCurrentUser(
+      'search',
+      query
+    )
+  );
+  const hostMetadata = hostResult.result;
+  if (!hostMetadata) {
+    return undefined;
+  }
+
+  const agent = await findAgent(metadataRequestContext, hostMetadata);
+
+  if (agent && !agent.active) {
+    throw Boom.badRequest('the requested endpoint is unenrolled');
+  }
+
+  const metadata = await enrichHostMetadata(
+    hostMetadata,
+    metadataRequestContext,
+    hostResult.queryStrategyVersion
+  );
+  return { ...metadata, query_strategy_version: hostResult.queryStrategyVersion };
+}
+
+async function findAgent(
+  metadataRequestContext: MetadataRequestContext,
+  hostMetadata: HostMetadata
+): Promise<Agent | undefined> {
+  try {
+    return await metadataRequestContext.endpointAppContextService
+      ?.getAgentService()
+      ?.getAgent(
+        metadataRequestContext.requestHandlerContext.core.savedObjects.client,
+        hostMetadata.elastic.agent.id
+      );
+  } catch (e) {
+    if (
+      metadataRequestContext.requestHandlerContext.core.savedObjects.client.errors.isNotFoundError(
+        e
+      )
+    ) {
+      metadataRequestContext.logger.warn(
+        `agent with id ${hostMetadata.elastic.agent.id} not found`
+      );
+      return undefined;
+    } else {
+      throw e;
+    }
+  }
+}
+
+export async function mapToHostResultList(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  queryParams: Record<string, any>,
+  hostListQueryResult: HostListQueryResult,
+  metadataRequestContext: MetadataRequestContext
+): Promise<HostResultList> {
+  const totalNumberOfHosts = hostListQueryResult.resultLength;
+  if (hostListQueryResult.resultList.length > 0) {
+    return {
+      request_page_size: queryParams.size,
+      request_page_index: queryParams.from,
+      hosts: await Promise.all(
+        hostListQueryResult.resultList.map(async (entry) =>
+          enrichHostMetadata(
+            entry,
+            metadataRequestContext,
+            hostListQueryResult.queryStrategyVersion
+          )
+        )
+      ),
+      total: totalNumberOfHosts,
+      query_strategy_version: hostListQueryResult.queryStrategyVersion,
```

Comment:
🕺 
Awesome. So the UI will use this new response property to conditionally show the KQL bar if version is `v2` 👍

Reply:
Yes